File: src/main/java/com/mechanitis/demo/boilerplate/example6/Lambdas.java
package com.mechanitis.demo.boilerplate.example6;

import java.util.function.BiConsumer;

@SuppressWarnings({"SameParameterValue", "unused"})
public class Lambdas {
    //<editor-fold desc="Fields">
    private final static RequestJava request = null;
    private final static ResponseJava response = null;
    //</editor-fold>

    public static void main(String[] args) {
        get("/home", (request, response) -> { /*do something */ } );
    }

    private static void get(String path, BiConsumer<RequestJava, ResponseJava> handler) {
        handler.accept(request, response);
    }


    private static class RequestJava {
    }

    private static class ResponseJava {
    }
}

File: src/main/java/com/mechanitis/demo/boilerplate/example6/Lambdas.kt
@file:Suppress("UNUSED_PARAMETER", "UNUSED_VARIABLE", "unused", "UNUSED_ANONYMOUS_PARAMETER")

package com.mechanitis.demo.boilerplate.example6

//val request = RequestKotlin
//val response = ResponseKotlin

    fun main(args: Array<String>) {
        get("/home") { request, response -> /*do something */ }
    }

    fun get(path: String, handler: (RequestKotlin, ResponseKotlin) -> Unit) {
//        handler(request, response)
    }

class RequestKotlin
class ResponseKotlin


